### ISSUE TITLE
AI factions are not firing their weapons

### ISSUE THREAD

**Issue by Malxe** [OuterDeepSpace/OuterDeepSpace#25]
AI factions (Renegade, Mutant, Pirate, E.D.E.N.) faction structures are armed with weapons and are supposed to shoot at players.  At present none are firing weapons at all, thus players take no damage.

**Comment by Ragora** [comment]
Good thing I worked out running a local server with the code on the repo here, debugging this would take a poking around in the server programming. The way the game handles exceptions in the server, there could be exceptions being thrown and being silenced for all we know.

Speaking of debugging the server -- what's the fastest time rate a galaxy can be run at? For debugging purposes, near-realtime would be very advantageous.

**Comment by Malxe** [comment]
The last time there was any serious third party testing of the server, the server did not internally handle turn processing rates, so we had to manually draft processTurn.py and run it via a crontab or similar scheduling method.  Going that route we were able to arrange turns to be processed as quickly as on a continuous basis.

However since the server now uses config.ini to specify the turn intervals I am not certain, but based on the way it is written, 1 minute turns would be feasible.

**Comment by Ragora** [comment]
1min should be okay at the very least, once I setup the test case I shouldn't have to worry so much about time anymore. Do something like declare war and put a scout in orbit of an AI system.

**Comment by Malxe** [comment]
Declare War would work for Renegade and Mutant factions.  Though for Pirate and EDEN, even though they too have "civilian ship passage" enabled by default they will still attempt to shoot at civilian player ships in orbit.

**Comment by Ragora** [comment]
I also wouldn't be sure what time frame to attach to this. The codebase isn't quite sane and I learned my lesson from #4 being a long spaghetti string leading back to session handling.

**Comment by Malxe** [comment]
Yeah, this is one of those "I have no idea what's wrong" type of issues, just like the one with galaxy news messages not sending.

**Comment by Delsorou** [comment]
This issue affects pirates too, previous PR did not address, reopening.
